provider/pkg/gen/nodejs-templates/helm/v3/helm.ts

```diff
@@ -0,0 +1,356 @@
+                pulumi.log.info("[Can't preview] all chart values must be known ahead of time to generate an " +
+                    "accurate preview.", this);
+            }
+        });
+
+        this.resources = allConfig.apply(cfg => {
+            return this.parseChart(cfg, releaseName, opts)
+        });
+    }
+
+    parseChart(config: ChartOpts | LocalChartOpts, releaseName: string, opts?: pulumi.ComponentResourceOptions) {
+        const blob = {
+            ...config,
+            releaseName,
+
+            toJSON() {
+                let obj: any = {};
+                for (const [key, value] of Object.entries(this)) {
+                    if (value) {
+                        switch(key) {
+                            case "apiVersions": {
+                                obj["api_versions"] = value;
+                                break;
+                            }
+                            case "caFile": {
+                                obj["ca_file"] = value;
+                                break;
+                            }
+                            case "certFile": {
+                                obj["cert_file"] = value;
+                                break;
+                            }
+                            case "fetchOpts": {
+                                obj["fetch_opts"] = value;
+                                break;
+                            }
+                            case "releaseName": {
+                                obj["release_name"] = value;
+                                break;
+                            }
+                            case "resourcePrefix": {
+                                obj["resource_prefix"] = value;
+                                break;
+                            }
+                            case "untardir": {
+                                obj["untar_dir"] = value;
+                                break;
+                            }
+                            default: {
+                                obj[key] = value;
+                            }
+                        }
+                    }
+                }
+                return obj
+            }
+        }
+
+        const jsonOpts = JSON.stringify(blob)
+
+        // Rather than using the default provider for the following invoke call, use the version specified
+        // in package.json.
+        let invokeOpts: pulumi.InvokeOptions = { async: true, version: getVersion() };
+
+        const promise = pulumi.runtime.invoke("kubernetes:helm:template", {jsonOpts}, invokeOpts);
+        return pulumi.output(promise).apply<{[key: string]: pulumi.CustomResource}>(p => yaml.parse(
+            {
+                resourcePrefix: config.resourcePrefix,
+                objs: p.result,
+                transformations: config.transformations || [],
+            },
+            { ...opts, parent: this }
+        ));
+    }
+}
+
+interface BaseChartOpts {
```

Comment:
They could probably be shared, but I think I'll leave them separate in case we want to diverge in the future given the upcoming deprecation plan for v2.